R/utilities.R

```diff
@@ -0,0 +1,346 @@
+library(tidyverse)
+library(behindbarstools)
+library(docstring)
+
+read_new_fac_info <- function(google_sheet_url = NULL) {
+    #' Read in UCLA facility data updates 
+    #'
+    #' Reads in new entries for the facility info sheet from Google Sheets.  
+    #'
+    #' @param google_sheet_url character string URL for the Google Sheet 
+    #'
+    #' @return data frame 
+    
+    if (is.null(google_sheet_url)) {
+        google_sheet_url <- "https://docs.google.com/spreadsheets/d/1tAhD-brnuqw0s55QXM-xYLPsyD-rNrqHbAVIbxSOMwI/edit#gid=1597803218"
+    }
+    
+    google_sheet_url %>% 
+        googlesheets4::read_sheet(sheet = "fac_data") %>% 
+        mutate(Name = behindbarstools::clean_fac_col_txt(Name, to_upper = TRUE)) %>% 
+        mutate_at(
+            vars(State, 
+                 Name, 
+                 Jurisdiction, 
+                 Designation, 
+                 Address, 
+                 City, 
+                 Zipcode, 
+                 County, 
+                 County.FIPS, 
+                 Security.Level, 
+                 Website), 
+            as.character) %>% 
+        mutate_at(
+            vars(HIFLD.ID, 
+                 Population.Feb, 
+                 Latitude, 
+                 Longitude,
+                 Capacity), 
+            as.numeric) %>% 
+        unique()
+}
+
+read_new_fac_spellings <- function(google_sheet_url = NULL) {
+    #' Read in UCLA facility spellings updates 
+    #'
+    #' Reads in new entries for the facility alternative spellings sheet from Google Sheets.  
+    #'
+    #' @param google_sheet_url character string URL for the Google Sheet 
+    #'
+    #' @return data frame 
+    
+    if (is.null(google_sheet_url)) {
+        google_sheet_url <- "https://docs.google.com/spreadsheets/d/1tAhD-brnuqw0s55QXM-xYLPsyD-rNrqHbAVIbxSOMwI/edit#gid=1597803218"
+    }
+    
+    google_sheet_url %>% 
+        googlesheets4::read_sheet(sheet = "fac_spellings") %>% 
+        mutate(xwalk_name_raw = behindbarstools::clean_fac_col_txt(xwalk_name_raw, to_upper = TRUE), 
```

Comment:
Thinking about redundancy between this and [read_fac_spellings](https://github.com/uclalawcovid19behindbars/behindbarstools/blob/master/R/read_fac_spellings.R) -- because we're going to clean the raw name every time we read in the fac_spellings sheet, does it make sense to remove the cleaning function here? Or perhaps better to remove the cleaning in read_fac_spellings, what do you think?

Reply:
Oh that's a good point! I think we would need to keep the cleaning in `read_fac_spellings` since the old spellings aren't cleaned, but you're definitely right that it's redundant to clean the string here AND whenever we pull the data. 

The options I can think of include (1) running `clean_fac_col_txt` once on the existing sheet, then dropping it from `read_fac_spellings` and trusting that all spellings moving forward will be cleaned in the adding process, (2) not cleaning it in the adding process and just cleaning it in the reading process, or (3) keeping it in both, since it's not that expensive of an operation? (I don't think, at least!) 

I'm perfectly indifferent across the options, whatever you think makes the most sense - or if there's other options you think would be better!

Reply:
@nmmarquez do you have any thoughts on this? I'm not really sure what is best practices, but now that I'm thinking more about it I am less worried about option 3

Reply:
I have been going with option 3 in the scraper. Idk if its ideal but the cost is cheap like Erika said and `clean_fac_col_txt` should not change the output of a character vector if its run multiple times.